clamscan/manager.c

```diff
@@ -1031,6 +1031,24 @@ static int scan_files(struct cl_engine *engine, const struct optstruct *opts, st
     return ret;
 }
 
+static cl_error_t check_if_cvd_outdated(const char *path, long long days)
+{
+    cl_error_t status;
+    time_t cvd_age;
+
+    if ((status = cl_cvdgetage(path, &cvd_age)) != CL_SUCCESS) {
+        logg(LOGG_ERROR, "%s\n", cl_strerror(status));
+        return status;
+    }
+
+    if (days * 86400 < cvd_age) {
+        logg(LOGG_ERROR, "Virus database is older than %lld days!\n", days);
+        return CL_ECVD;
+    }
+
+    return CL_SUCCESS;
+}
+
```

Comment:
If you want to put it in `common` you can minimize duplicate code when adding this feature to `clamd`.  Maybe in `common/misc.c` / `.h`.